library/ssl_misc.h

```diff
@@ -582,10 +582,17 @@ struct mbedtls_ssl_handshake_params
 #endif /* MBEDTLS_SSL_PROTO_TLS1_3 */
 
 #if defined(MBEDTLS_SSL_CLI_C)
-    /*!<  Number of Hello Retry Request messages received from the server.  */
+    /** Number of Hello Retry Request messages received from the server.  */
     int hello_retry_request_count;
 #endif /* MBEDTLS_SSL_CLI_C */
 
+#if defined(MBEDTLS_SSL_SRV_C)
+    /** selected_group of key_share extension in HelloRetryRequest message. */
+    uint16_t hrr_selected_group;
+    /** selected_group of key_share extension in ClientHello message. */
+    uint16_t selected_group;
```

Comment:
Why not share `offered_group_id`?

Reply:
```suggestion
```
hrr_selected_group has been defined. And comments for `selected_group` is not correct. Please remove it and only keep hrr_selected_group. 
Again, I suggest totally remove it and empty `ssl_tls13_parse_groups.ext` function in this PR.

Reply:
> Why not share `offered_group_id`?
Ronald advice me to refer your PR to define one new field to replace  offered_group_id

Reply:
> 

I just try to remove all new added hrr related field, so one new field is added. Anyway,  I think it's fine to change back to hrr_selected_group,

Reply:
I think you misunderstand it . Not replace `offered_group_id`, That's to add `hrr_selected_group`.
That's relative many thing. If the feature is added in this PR. I think more test should be added here. 
To reduce the scope of this PR, I suggest empty the function.

Reply:
How aboout your opinion? @ronald-cron-arm

Reply:
There are several things here. First there should be only `selected_group` or `hrr_selected_group` but not both. Currently `hrr_selected_group` is introduced in this PR but not used. Sticking here to `selected_group` is fine by me. Second, to keep or not this function in this PR. I think it makes sense to keep it. Even if we do not use the result of the parsing of this function (`selected_group`) in this PR, the parsing it does is tested by the tests added by this PR.

Reply:
Comments in line 592 is wrong. selected_group or hrr_selected_group is used for HRR message , not ClientHello message.

Reply:
I prefer keep `hrr_selected_group`. it is easy understand what's the purpose of the variable.